userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1580,6 +1633,125 @@ int32_t scap_bpf_get_stats(struct scap_engine_handle engine, OUT scap_stats* sta
 	return SCAP_SUCCESS;
 }
 
+struct scap_stats_v2* scap_bpf_get_stats_v2(struct scap_engine_handle engine, uint32_t flags, OUT uint32_t* nstats, OUT int32_t* rc)
+{
+	struct bpf_engine *handle = engine.m_handle;
+	int ret;
+	int fd;
+	int i = 0; // offset in stats
+	*nstats = 0;
+	scap_stats_v2* stats = handle->m_stats;
+	if (!stats)
+	{
+		*rc = SCAP_FAILURE;
+		return NULL;
+	}
+
+	if ((flags & PPM_SCAP_STATS_KERNEL_COUNTERS))
+	{
+		/* KERNEL SIDE STATS COUNTERS */
+		for(uint32_t stat =  0;  stat < BPF_MAX_KERNEL_COUNTERS_STATS; stat++)
+		{
+			stats[stat].type = STATS_VALUE_TYPE_U64;
+			stats[stat].flags = PPM_SCAP_STATS_KERNEL_COUNTERS;
+			stats[stat].value.u64 = 0;
+			strlcpy(stats[stat].name, bpf_kernel_counters_stats_names[stat], STATS_NAME_MAX);
```

Comment:
(a general comment about all the engines)

We have to copy the names over and over, so how about we:
1. copy them just once, on init
2. change the method signature to return a `const struct scap_stats_v2*`
?

EDIT: sigh, the stats list is dynamic, based on flags, no can do. But I still feel a `const` would be appropriate.

Reply:
Hmm for some subsets of the flags bitmap values we could simply return different offsets into the stats struct (and different sizes)

Reply:
Perhaps keep in mind and consider for a future adjustment?

Reply:
Yup, we don't need it exactly on day 1